server/handlers/handlers.go

```diff
@@ -121,20 +121,25 @@ func (ah *ActivityHandler) Handle(activity msteams.Activity) error {
 
 func (ah *ActivityHandler) HandleLifecycleEvent(event msteams.Activity) {
 	if !ah.checkSubscription(event.SubscriptionID) {
+		ah.plugin.GetMetrics().ObserveLifecycleEvent(event.LifecycleEvent, metrics.DiscardedReasonFailedSubscriptionCheck)
 		return
 	}
 
 	if event.LifecycleEvent == "reauthorizationRequired" {
 		expiresOn, err := ah.plugin.GetClientForApp().RefreshSubscription(event.SubscriptionID)
 		if err != nil {
 			ah.plugin.GetAPI().LogError("Unable to refresh the subscription", "error", err.Error())
-		} else {
-			ah.plugin.GetMetrics().ObserveSubscription(metrics.SubscriptionRefreshed)
-			if err = ah.plugin.GetStore().UpdateSubscriptionExpiresOn(event.SubscriptionID, *expiresOn); err != nil {
-				ah.plugin.GetAPI().LogError("Unable to store the subscription new expiry date", "subscriptionID", event.SubscriptionID, "error", err.Error())
-			}
+			ah.plugin.GetMetrics().ObserveLifecycleEvent(event.LifecycleEvent, metrics.DiscardedReasonFailedToRefresh)
+			return
+		}
+
+		ah.plugin.GetMetrics().ObserveSubscription(metrics.SubscriptionRefreshed)
```

Comment:
Note that we observe the refreshed subscription at the same rate we observe the lifecycle event. There's a possibility one of these metrics is redundant, but keeping for now.